src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder2.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    // We use a subclass of SwitchBinder for the pattern-matching switch statement until we have completed
+    // a totally compatible implementation of switch that also accepts pattern-matching constructs.
+    internal partial class PatternSwitchBinder2 : SwitchBinder
+    {
+        internal PatternSwitchBinder2(Binder next, SwitchStatementSyntax switchSyntax) : base(next, switchSyntax)
+        {
+        }
+
+        /// <summary>
+        /// When pattern-matching is enabled, we use a completely different binder and binding
+        /// strategy for switch statements. Once we have confirmed that it is totally upward
+        /// compatible with the existing syntax and semantics, we will merge them completely.
```

Comment:
>we will merge them completely [](start = 63, length = 29)

Should we have a PROTOTYPE comment for this?

Reply:
This doc comment is for the reader to understand this property, not to track work. The relevant work is tracked in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222)

---
In reply to: [156523029](https://github.com/dotnet/roslyn/pull/23209#discussion_r156523029) [](ancestors = 156523029)